app/webpacker/components/Tickets/TicketCommentCreate.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Button, Modal } from 'semantic-ui-react';
+import { useMutation } from '@tanstack/react-query';
+import MarkdownEditor from '../wca/FormBuilder/input/MarkdownEditor';
+import useInputState from '../../lib/hooks/useInputState';
+import createComment from './api/createComment';
+import Loading from '../Requests/Loading';
+import Errored from '../Requests/Errored';
+
+export default function TicketCommentCreate({
+  open, onClose, ticketId, currentStakeholder, refetchComments,
+}) {
+  const [comment, setComment] = useInputState();
+
+  const {
+    mutate: createCommentMutation,
+    isPending,
+    isError,
+  } = useMutation({
+    mutationFn: createComment,
+    onSuccess: () => {
+      setComment('');
+      refetchComments();
+      onClose();
+    },
+  });
```

Comment:
```suggestion
const queryClient = useQueryClient();

const {
  mutate: createCommentMutation,
  isPending,
  isError,
} = useMutation({
  mutationFn: createComment,
  onSuccess: (returnValueOfMutationFn, parametersPassedToMutationFn) => {
    setComment('');
    queryClient.setQueryData(['my-query-key', myQueryId, someOtherId], (previousData) => [...previousData, parametersPassedToMutationFn]);
    onClose();
  },
});
```

Reply:
Essentially, you're writing into the client-side cache storage, which lets you completely drop the `refetchComments` part and save one "fetch from backend -> set loading state -> display" cycle. In particular, it saves network traffic and database load. Hooray!

Reply:
(Note that in the future, you may run across a very special edge case where it is necessary to manually refetch. It's not 100% forbidden to do that. But in at least 8/10 cases, like simple "add a comment" or "manipulate an existing comment", you can easily modify the data in-place, without having to refetch the exact same comment which you just added anyways)

Reply:
Reference: https://tanstack.com/query/latest/docs/framework/react/guides/mutations#mutation-side-effects